File: Studia/Projekt_Hurtowni_c++/Drukarka.cpp
#include "Classes/Drukarka.hpp"
#include <fstream>

Drukarka::Drukarka(string p, string m, bool k, string t, int c) : Produkt(p,m,c)
{
	kolor = k;
	typ = t;
}

void Drukarka::wyswietl_produkt()
{
	cout<<"*Drukarka*	Producent: "<<producent<<", Model: "<<model<<", kolor: "<<kolor<<", typ: "<<typ<<", cena: "<<cena;
}

void Drukarka::zapisz_produkt()
{
	fstream plik;
	plik.open("Drukarka.txt", ios::app);
	if(plik.good())
	{
		plik<<producent<<" "<<model<<" "<<kolor<<" "<<typ<<" "<<cena<<" ";
	}
	plik.close();
}

File: Studia/Projekt_Hurtowni/src/projekt_hurtowni/Drukarka.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package projekt_hurtowni;
import java.io.*;
/**
 *
 * @author Bartek
 */
public class Drukarka extends Produkt {
    String producent;
    String model;
    String kolor;
    String typ;
    int cena;
    
    public Drukarka(String p, String m, String k, String t, int c)
    {
     producent = p;
     model = m;
     kolor = k;
     typ = t;
     cena = c;   
    }
    @Override
    public void wyswietl_produkt()
    {
           System.out.print("*Drukarka*   producent: " + producent + ", model: "
           + model + ",  kolor: " + kolor + ", typ: " + typ + ", cena: " + cena);
    }
    /**
     *
     * @throws java.io.FileNotFoundException
     */
    @Override
    public void zapisz_produkt() throws FileNotFoundException
    {
        try (PrintWriter zapis = new PrintWriter(new FileOutputStream(new File("Drukarka.txt"), true))) 
        {
            zapis.append(producent + ":" + model + ":" + kolor + ":" + typ + ":" + cena + ":");
            zapis.close();
        }
    }
}
